#summary Description and instructions for GSRf.

= GSRf =

JPrIME-GSRf is a phylogenetic application for reconciling an unknown guest tree and a known host tree by means of duplication events and loss events. Typically, the guest tree corresponds to a gene tree, and the host tree corresponds to a species tree. JPrIME-GSRf relies on Bayesian inference using an MCMC framework. Thus, it will yield a posterior distribution of guest tree topologies and remaining parameters.

*Input:*
  * a file with a dated host tree.
  * a file with multialignment of guest tree leaf sequences.
  * a file with a simple map relating the guest tree leaves to host tree leaves.

*Output:*
  * a file with samples drawn from the posterior distribution.

== Running ==
 
=== Host tree ===
Obtain a dated species tree in Newick format. The divergence times may be specified as branch lengths, e.g.:

{{{((A:0.4,B:0.4):0.6,C:1.0):0.5;}}}

Alternatively, one can specify absolute times using special formatting:

{{{((A[&&PRIME NT=0.0],B[&&PRIME NT=0.0])[&&PRIME NT=0.4],C[&&PRIME NT=0.0])[&&PRIME NT=1.0][&&PRIME TT=0.5];}}}

The host tree must be ultrametric, with time 0 at the leaves and >0 at the root, and a "stem" of length >0 (above, this  is set to 0.5). The scale of the dates can be any unit: *they will be rescaled prior to running internally so that the root time is 1.0, and the output will refer to this scaling*, and not the original.

A good resource for divergence times is [http://www.timetree.org]. If you cannot find references to reliable divergence times, you will need to date the tree yourself. Typically, this is achieved by obtaining a set of putative core gene families (e.g. monocopy putative orthologs), aligning these, then feeding the concatenated super-alignment into a program which estimates divergence times. 